sd/etcd/registrar.go

```diff
@@ -43,6 +67,22 @@ func (r *Registrar) Register() {
 	} else {
 		r.logger.Log("action", "register")
 	}
+	if r.service.TTL == nil {
+		return
+	}
+	r.quit = make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-r.quit:
+				return
+			case <-time.After(r.service.TTL.Heartbeat):
```

Comment:
This will generate a lot of unnecessary garbage. It's better to use a time.Ticker, constructed outside of the for loop.